File: models/Gpt-3.5-turbo/cpp/code/problems/introductory_problems/Creating_Strings.cpp

#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

int main() {
    string s;
    cin >> s;

    // Sort the string lexicographically
    sort(s.begin(), s.end());

    vector<string> permutations;

    // Generate all permutations using next_permutation
    do {
        permutations.push_back(s);
    } while (next_permutation(s.begin(), s.end()));

    // Print the number of strings
    cout << permutations.size() << endl;

    // Print the strings in alphabetical order
    for (const string& perm : permutations) {
        cout << perm << endl;
    }

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/introductory_problems/Creating_Strings.py

def generate_permutations(s):
    # Set to store unique permutations
    permutations = set()

    def backtrack(s, curr):
        # Base case: if the current permutation is of length n
        if len(curr) == len(s):
            permutations.add(curr)
            return
        
        # Recursive case: try all possible characters at the current position
        for i in range(len(s)):
            if s[i] not in curr:
                # Add the character to the current permutation
                curr += s[i]
                # Recursively generate permutations
                backtrack(s, curr)
                # Remove the character for backtracking
                curr = curr[:-1]

    # Start backtracking
    backtrack(s, "")

    return permutations

# Driver code
s = input().strip()
permutations = generate_permutations(s)

# Print the number of strings and the strings in alphabetical order
print(len(permutations))
for p in sorted(permutations):
    print(p)
